### PR TITLE
Disable various missions for integration tests

### PR DESCRIPTION
**Pull request by warp-core**
**Bugfix:**

## Fix Details
Round 3?
Disabled six invisible landing missions that offer unconditionally one a new pilot.
I believe that handling these missions all at once was causing the game to miss one of the key inputs (this wouldn't happen to a real player because there is likely a large enough delay for the game to finish the mission handling and a real player is unlikely to only hold down a key for a single frame, similarly, this issue doesn't present when running the integration test with `--debug` because the test sequence is executed much more slowly and so there is ample time for hte missions to be handled.)

An alternative to disabling these missions would be to add conditions to the test data save files that will prevent them from offering.

I have also removed an inappropriate call to "Open Menu" in the "Loading and Saving" test; the "Load First Savegame" sequence hits 'escape' for itself, so calling "Open Menu" immediately prior causes that helper to exit the menu and try to load a save on the planet panel, which isn't going to work. It ends up in the bank.

~~Round 2:~~
~~I'm not entirely sure how or why (and discussing my speculation is probably going to be rather long), but disabling this mission seems to allow the "Loading and Saving" integration test to run as it previously did.~~
~~I still think the integration test isn't quite right, as mentioned in #7955, but it at least now goes back to working as it was before the Gegno merge. One step at a time.~~

## Testing Done
Ran the test locally. It passed. We'll see what happens here.

## Save File
Integration tests have their own save files.

### CODE REVIEW

**Comment by warp-core** [comment]
It passed! Let's merge?

**Comment by Zitchas** [comment]
I've always figured that the whole lot of those missions were a problem. We should have means of ensuring an NPC is just always present in a system without needing a whole mission cluttering up the save file to do it. And any conversation/dialog required could be handled by a mission that triggers when entering the system.

**Comment by warp-core** [comment]
> I've always figured that the whole lot of those missions were a problem. We should have means of ensuring an NPC is just always present in a system without needing a whole mission cluttering up the save file to do it. And any conversation/dialog required could be handled by a mission that triggers when entering the system.

You're welcome to provide that functionality.

**Comment by Zitchas** [comment]
Oh, I would if I could. Definitely would if I could.

**Comment by warp-core** [comment]
Ok, I think we're done?

File: tests/integration/config/plugins/integration-tests/data/tests/tests_save_load.txt [hunk 1]
```diff
@@ -43,6 +43,12 @@ test "Loading and Saving"
 		inject "Three Earthly Barges"
 		call "Load First Savegame"
 		call "Depart"
+		apply
+			"test: Depart: 0 zoom_restore_counter" = "test: Depart: zoom_restore_counter"
+			"test: 0 departed 1" = 1
+			"test: year" = year
+			"test: month" = month
+			"test: day" = day
```
Comment (warp-core) [inline tests/integration/config/plugins/integration-tests/data/tests/tests_save_load.txt:51]:
```suggestion
```

File: tests/integration/config/plugins/integration-tests/data/tests/tests_save_load.txt [hunk 2]
```diff
@@ -69,13 +80,28 @@ test "Loading and Saving"
 		label notReturned1
 		branch notReturned1
 			not "flagship planet: Earth"
+		apply
+			"test: 2 returned to Earth 1" = 1
+			"test: year" = year
+			"test: month" = month
+			"test: day" = day
```
Comment (warp-core) [inline tests/integration/config/plugins/integration-tests/data/tests/tests_save_load.txt:87]:
```suggestion
```

File: tests/integration/config/plugins/integration-tests/data/tests/tests_save_load.txt [hunk 3]
```diff
@@ -69,13 +80,28 @@ test "Loading and Saving"
...
 			year == 3013
 			month == 11
 			day == 19
-		call "Open Menu"
 		call "Save To Default Snapshot"
+		apply
+			"test: 3 saved to default snapshot" = 1
+			"test: year" = year
+			"test: month" = month
+			"test: day" = day
```
Comment (warp-core) [inline tests/integration/config/plugins/integration-tests/data/tests/tests_save_load.txt:97]:
```suggestion
```

File: tests/integration/config/plugins/integration-tests/data/tests/tests_save_load.txt [hunk 4]
```diff
@@ -69,13 +80,28 @@ test "Loading and Saving"
...
 		call "Depart"
+		apply
+			"test: Depart: 1 zoom_restore_counter" = "test: Depart: zoom_restore_counter"
+			"test: 4 departed again" = 1
+			"test: year" = year
+			"test: month" = month
+			"test: day" = day
```
Comment (warp-core) [inline tests/integration/config/plugins/integration-tests/data/tests/tests_save_load.txt:104]:
```suggestion
```

File: tests/integration/config/plugins/integration-tests/data/tests/tests_save_load.txt [hunk 5]
```diff
@@ -92,13 +118,22 @@ test "Loading and Saving"
...
 			year == 3013
 			month == 11
 			day == 21
 		call "Open Menu"
 		call "Load First Savegame"
+		apply
+			"test: 6 loaded first savegame" = 1
 		call "Depart"
+		apply
+			"test: 7 departed one more time" = 1
```
Comment (warp-core) [inline tests/integration/config/plugins/integration-tests/data/tests/tests_save_load.txt:136]:
```suggestion
```

File: tests/integration/config/plugins/integration-tests/data/tests/tests_save_load.txt (hunk 1 above)
Comment (Hurleveur) [reply tests/integration/config/plugins/integration-tests/data/tests/tests_save_load.txt:51]: I think it doesnt hurt to keep them, if this test was so hard to debug it was partly because of lack of test conditions
That said most of these could be put in an independant function?

File: tests/integration/config/plugins/integration-tests/data/disabling troublesome missions.txt [hunk 6]
```diff
@@ -0,0 +1,6 @@
+disable mission "Conversation on Sayaiban"
+disable mission "Drak guarding Korath space"
+disable mission "Drak guarding Sayaiban"
+disable mission "Drak guarding Void Sprites"
+disable mission "Relic of the Past"
+disable mission "Stationed Augen"
```
Comment (Hurleveur) [inline tests/integration/config/plugins/integration-tests/data/disabling troublesome missions.txt:6]: We can do this?? We should also disable landing on ruin mission then I had to set a condition for it being offered everywhere
Could be in another PR though

File: tests/integration/config/plugins/integration-tests/data/tests/tests_save_load.txt [hunk 7]
```diff
@@ -57,6 +63,11 @@ test "Loading and Saving"
 		label notAlpha1
 		branch notAlpha1
 			not "flagship system: Alpha Centauri"
+		apply
+			"test: 1 entered A Cen 1" = 1
+			"test: year" = year
+			"test: month" = month
+			"test: day" = day
```
[the remark this answers is not in the document]
Comment (Hurleveur) [reply tests/integration/config/plugins/integration-tests/data/tests/tests_save_load.txt:70]: I'm confused why are they all named in the same way? Shouldnt it be year1 year2 etc?
We could also just keep the last one I guess

File: tests/integration/config/plugins/integration-tests/data/tests/tests_save_load.txt [hunk 8]
```diff
@@ -92,13 +118,22 @@ test "Loading and Saving"
...
 		assert
 			year == 3013
 			month == 11
 			day == 21
 		call "Open Menu"
 		call "Load First Savegame"
+		apply
+			"test: 6 loaded first savegame" = 1
```
[the remark this answers is not in the document]
Comment (Hurleveur) [reply tests/integration/config/plugins/integration-tests/data/tests/tests_save_load.txt:133]: I think that this one isnt that required because we should instead get where exactly the test stoped

**Review by Hurleveur — CHANGES_REQUESTED** [review]
I think that this reveals a big issue with integration tests: we dont know where they fail.
The simplest way to do this PR (that needs to fix master asap) is to assume that will be added later.
Put all the test: year etc in the same partial test to call it and remove checkpoint checks. Then, in another PR we need to give the exact line that failed instead of a number.

**Review by quyykk — APPROVED** [review]
lgtm. @warp-core do you want to do anything else?

**Comment by Hurleveur** [comment]
It could always be done in another PR tbf

**Comment by warp-core** [comment]
I think this is ready. Disabling Landing On Ruin and removing the condition from the test data save files is its own thing.
@quyykk
